### PR TITLE
Marker icons

### PR DESCRIPTION
**Pull request by Almightywesley**
Changes:
i added different types of markers in the app
How to test: 
 1. make a memory
 2. chose a marker

### CODE REVIEW

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryDetailActivity.java [hunk 1]
```diff
@@ -26,6 +28,8 @@
     Intent shareIntent;
     Memory memory;
     ViewPager2 viewPager2;
+    GoogleMap mGoogleMap;
+    MapView mMapView;
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryDetailActivity.java:41]: So, it seems like the only differences in this file (`MemoryDetailActivity`) are 2 new unused variables and a couple empty. Please undo these changes.

**Review by AnassM06 — COMMENTED** [review]
wesley you have to fix the conflicts in the code

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java [hunk 2]
```diff
@@ -107,7 +108,8 @@ protected void onCreate(Bundle savedInstanceState) {
                     new Date(calendar.getTimeInMillis()),
                     "",
                     "",
-                    media
+                    media,
+                    R.drawable.ic_map_adefaultl
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: There seems to be a typo here. It should be `ic_map_adefault` and not `ic_map_adefaultl`

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java [hunk 3]
```diff
@@ -214,6 +216,14 @@ public void onClick(View v) {
                 startActivityForResult(i, LOCATION_EDIT);
             }
         });
+        spinner = findViewById(R.id.Marker_selector);
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: this `findViewById` should be with the others (near line `77`). Otherwise it'll be harder to find.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java [hunk 4]
```diff
@@ -214,6 +216,14 @@ public void onClick(View v) {
...
+        spinner.setOnItemSelectedListener(new SimpleOnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                super.onItemSelected(parent, view, position, id);
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: `super.onItemSelected(...);` can be removed because it is not necessariy. `SimpleOnItemSelectedListener` has no code in the implementation so it does literally nothing.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java [hunk 5]
```diff
@@ -243,6 +253,145 @@ public void onClick(View v) {
...
+                foundIcon = R.drawable.ic_map_tourist;
+                break;
+            case 41:
+                foundIcon = R.drawable.ic_map_want_to_go;
+                break;
+            case 42:
+                foundIcon = R.drawable.ic_map_world;
+                break;
+            default:
+                foundIcon = R.drawable.ic_map_adefaultl;
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: Are you 100% sure that all these icons are on the right position?

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/data/room/AppDatabase.java [hunk 6]
```diff
@@ -12,7 +12,7 @@
  * Created by maartendegoede on 23/04/2019.
  * Copyright © 2019 Anass El Mahdaoui, Hicham El Marzgioui, Wesley de Man, Maarten de Goede all rights reserved.
  */
-@Database(entities = {Memory.class, Media.class, Image.class, Video.class}, version = 1)
+@Database(entities = {Memory.class, Media.class, Image.class, Video.class}, version = 2)
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/data/room/AppDatabase.java:19]: I'll figure out if this version update is really necessary and how to handle it. Don't merge anything just yet.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java [hunk 7]
```diff
@@ -149,18 +151,29 @@ public void onInfoWindowClick(Marker marker) {
...
     }
+    private BitmapDescriptor bitmapDescriptorFromVector(Context context, int vectorResId) {
+        Drawable vectorDrawable = ContextCompat.getDrawable(context, vectorResId);
+        vectorDrawable.setBounds(0, 0, vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight());
+        Bitmap bitmap = Bitmap.createBitmap(vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight(),
+                Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(bitmap);
+        vectorDrawable.draw(canvas);
+        return BitmapDescriptorFactory.fromBitmap(bitmap);
+    }
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java]: I think you should move this code to the `Memory` class with a function called `getTypeBitmap` or something similar. Then you can do `memory.getTypeBitmap(context)` on any page. 
It should not have a `vectorResId` parameter obviously, as it should use the res id already set in the memory.

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -32,6 +32,7 @@
     <string name="action_ok">Got it!</string>
     <string name="action_apply_theme">Apply Theme</string>
     <string name="action_change_memory_location">Change location</string>
+    <string name="action_change_memory_marker_restaurant"></string>
```
Comment (MrTheGood) [inline app/src/main/res/values/strings.xml]: This string resource does not have a value and isn't used anywhere. It should be removed.

**Review by MrTheGood — CHANGES_REQUESTED** [review]
I will not approve it yet, as there are still some problems. But overall, nice work!

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java (hunk 2 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java (hunk 3 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java (hunk 4 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: done

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java (hunk 5 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryEditActivity.java]: yes!

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java (hunk 7 above)
Comment (Almightywesley) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java]: okidoki

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (Almightywesley) [reply app/src/main/res/values/strings.xml]: oki

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryDetailActivity.java (hunk 1 above)
Comment (MrTheGood) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryDetailActivity.java:41]: Seems fixed.

**Comment by MrTheGood** [comment]
Don't forget to add colours to the icons and to select the correct value when edit is selected.

**Review by MrTheGood — APPROVED** [review]
Don't forget to add colours to the icons and to select the correct value when edit is selected.

**Review by MrTheGood — CHANGES_REQUESTED** [review]
Don't forget to add colours to the icons and to select the correct value when edit is selected.

**Review by Hicham010 — COMMENTED** [review]
Your code looks great ?

**Review by MrTheGood — CHANGES_REQUESTED** [review]
The icons are missing colours. 
Otherwise, really nice work!

**Comment by MrTheGood** [comment]
Rode kaart.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/data/room/AppDatabase.java [hunk 9]
```diff
@@ -12,8 +16,15 @@
...
 @TypeConverters({Converters.class})
 public abstract class AppDatabase extends RoomDatabase {
     public abstract MemoryDao memoryDao();
+
+    public static final Migration MIGRATION_1_2 = new Migration(1, 2) {
+        @Override
+        public void migrate(@NonNull SupportSQLiteDatabase database) {
+            database.execSQL("ALTER TABLE Memory ADD COLUMN memoryTypeIconId INTEGER NOT NULL DEFAULT '" + R.drawable.ic_map_adefaultl + "'");
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/data/room/AppDatabase.java]: It still says `ic_map_adefaultl` here instead of `ic_map_adefault`. Please fix that.

**Review by MrTheGood — CHANGES_REQUESTED** [review]

**Review by MrTheGood — CHANGES_REQUESTED** [review]
The text for the viewpager indicator should be white. 
It also should have a 8dp margin on all sides, otherwise it's too close to the image sides.

**Review by MrTheGood — APPROVED** [review]

**Comment by MrTheGood** [comment]
Looks good! Nice work!
I did some last change myself because this PR has been going on for really long.

**Review by Hicham010 — APPROVED** [review]
Nice work wesley. I can see that you have implemented my feedback. Keep up the good work!!

**Review by MrTheGood — APPROVED** [review]
